### PR TITLE
feat: pass ef_test stSLoadTest

### PR DESCRIPTION
**Pull request by belfortep**
Closes https://github.com/lambdaclass/evm_mlir/issues/370
Closes https://github.com/lambdaclass/evm_mlir/issues/227

This PR fixs the ef_test for stSLoadTest, adding the warm/cold gas cost

### CODE REVIEW

File: src/syscall.rs [hunk 1]
```diff
@@ -1104,7 +1111,7 @@ impl<'c> SyscallContext<'c> {
             );
             engine.register_symbol(
                 symbols::STORAGE_READ,
-                SyscallContext::read_storage as *const fn(*const c_void, *const U256, *mut U256)
+                SyscallContext::read_storage as *const fn(*mut c_void, *const U256, *mut U256)
```
Comment (juanimedone) [inline src/syscall.rs:1114]: Is this change needed?

Comment (belfortep) [reply src/syscall.rs:1114]: Oh you are right, the `self` value isn't modified, it could be `*const` and the syscall receive a `&self`

Comment (belfortep) [reply src/syscall.rs:1114]: Never mind, the `*mut` is needed because the syscall use the `read_storage` function that needs a mut reference, I think it works either way but maybe it could generate undefined behaviour with the `*const`

Comment (juanimedone) [reply src/syscall.rs:1114]: Ok yes, we should leave it with `*mut`.

**Review by juanimedone — APPROVED** [review]
Looks good to me!

**Review by MegaRedHand — APPROVED** [review]
Great work!!
